libs/s25main/mapGenerator/Algorithms.h

```diff
@@ -28,6 +28,17 @@
 
 namespace rttr { namespace mapGenerator {
 
+    template<template<typename...> class C1, typename... T1, template<typename...> class C2, typename... T2, typename E>
+    std::vector<E> join(const C1<C2<E, T2...>, T1...>& container)
```

Comment:
There might be a better place for crazy template functions.

Reply:
This is indeed a bit crazy... Let's see, you have a container of containers here and need the type of the innermost element. So:
```
template<typename T>
auto join(const T& container){
  using E = typename T::value_type::value_type;
```

Is this function performance relevant? If so a sum of the sizes passed to reserve would help.

Reply:
It's not really performance relevant, at least not the way it is currently used (but I get your point). 

Just a nice challenge to join all elements in a container of containers I thought. When used it only saves a few lines of code in another place, was just proud I got it compiled lol. 

Might be a bit simpler to understand with more meaningful template type names I guess.